Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -4,52 +4,107 @@ import Foundation
 /// A global singleton which holds base configuration for the OpenFeature library.
 /// Configuration here will be shared across all ``Client``s.
 public class OpenFeatureAPI {
-    private var _provider: FeatureProvider? {
-        get {
-            providerSubject.value
-        }
-        set {
-            providerSubject.send(newValue)
-        }
-    }
-    private var _context: EvaluationContext?
+    private let eventHandler = EventHandler()
+    private let queue = DispatchQueue(label: "com.openfeature.providerDescriptor.queue")
+
+    private(set) var providerSubject = CurrentValueSubject<FeatureProvider?, Never>(nil)
+    private(set) var evaluationContext: EvaluationContext?
+    private(set) var providerStatus: ProviderStatus = .notReady
     private(set) var hooks: [any Hook] = []
-    private var providerSubject = CurrentValueSubject<FeatureProvider?, Never>(nil)
 
     /// The ``OpenFeatureAPI`` singleton
     static public let shared = OpenFeatureAPI()
 
     public init() {
     }
 
-    public func setProvider(provider: FeatureProvider) {
-        self.setProvider(provider: provider, initialContext: nil)
+    /**
+    Set provider and calls its `initialize` in a background thread.
+    Readiness can be determined from `getState` or listening for `ready` event.
+    */
+    public func setProvider(provider: FeatureProvider, initialContext: EvaluationContext?) {
+        queue.async {
+            Task {
+                await self.setProviderInternal(provider: provider, initialContext: initialContext)
+            }
+        }
```

Comment:
One limitation is the case where the Provider vendor wants to offer a way to synchronously make the Provider ready (for example because the Provider offers local-resolve or cached flags data). Now OF only offers two APIs to set a provider, and both require "awaiting": for an async function or a `.ready` event to arrive.